### PR TITLE
cleaning node controller from code for unsupported versions

### LINKED ISSUE #52356: clean node controller from code that supports version prior to 1.5
As discussed on slack with @gmarek, we might want to clean the node controller from code that ensures backward compatibility for version 1.4 and earlier ones.

/sig node

### PR DESCRIPTION
**Pull request by yastij**
**What this PR does / why we need it**:

**Which issue this PR fixes**: fixes #52356 

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/controller/node/node_controller.go [hunk 1]
```diff
@@ -1173,33 +1173,4 @@ func (nc *Controller) maybeDeleteTerminatingPod(obj interface{}) {
 	if pod.DeletionTimestamp == nil {
 		return
 	}
-
```
Comment (gmarek) [inline pkg/controller/node/node_controller.go]: After this change the whole function is a no-op, and should be removed. This is rather big change, so I'd rather do it for 1.9 to give it more time to bake.

Comment (yastij) [reply pkg/controller/node/node_controller.go]: +1
This shouldn't be merged for 1.8

**Comment by k82cn** [comment]
duplicated with https://github.com/kubernetes/kubernetes/pull/48996 ?

**Comment by yastij** [comment]
Indeed, closing this one in favor of #48996

**Comment by gmarek** [comment]
Yeah, but the former is huge so it'll be really really hard to find someone to review all the changes. More localized PRs would be better (especially if we end up reverting some).

**Comment by k82cn** [comment]
+1

**Comment by yastij** [comment]
@gmarek - ping for lgtm/approval

**Pull request merged**